packages/core/src/server/socketServer.ts

```diff
@@ -166,18 +167,36 @@ export class SocketServer {
       return null;
     }
 
+    // web-infra-dev/rspack#6633
+    const newEntryPointsLength = Object.values(stats.entrypoints ?? {}).reduce(
+      (prev, curr) => {
+        return prev + curr.chunks.length;
+      },
+      0,
+    );
+
+    const shouldReload =
+      this.entryPointsLength !== undefined &&
+      newEntryPointsLength !== this.entryPointsLength;
+    this.entryPointsLength = newEntryPointsLength;
+
     const shouldEmit =
       !force &&
       stats &&
       !stats.errorsCount &&
       stats.assets &&
-      stats.assets.every((asset: any) => !asset.emitted);
+      stats.assets.every((asset: any) => {
```

Comment:
Why changed `stats.assets.every((asset: any) => !asset.emitted);` to current function..